packages/hir/src/expressions.candy

```diff
@@ -1,182 +1,178 @@
-use ..declarations
+/*use ..declarations
 use ..types
 
-public data class ExpressionUri<E: Expression> {
-  public let declaration: Constructor | Function | Property
+public data class HirExpressionUri<E: HirExpression> {
+  public let declaration: HirConstructor | HirFunction | HirProperty
   public let index: Int
 }
 
 
-public trait Expression {
-  public let uri: ExpressionUri<Expression>
+public trait HirExpression {
+  public let uri: HirExpressionUri<HirExpression>
 }
 
 
 
-public class BoolExpression {
-  public let uri: ExpressionUri<BoolExpression>
+public class HirBoolExpression {
+  public let uri: HirExpressionUri<HirBoolExpression>
   public let value: Bool
 }
-impl BoolExpression: Expression
+impl HirBoolExpression: HirExpression
 
-public class IntExpression {
-  public let uri: ExpressionUri<IntExpression>
+public class HirIntExpression {
+  public let uri: HirExpressionUri<HirIntExpression>
   public let value: Int
 }
-impl IntExpression: Expression
+impl HirIntExpression: HirExpression
 
 
-public class StringExpression {
-  public let uri: ExpressionUri<StringExpression>
-  public let parts: List<StringPart>
+public class HirStringExpression {
+  public let uri: HirExpressionUri<HirStringExpression>
+  public let parts: List<HirStringPart>
 }
-impl StringExpression: Expression
+impl HirStringExpression: HirExpression
 
-trait StringPart
-impl StringPart: Expression
+trait HirStringPart
 
-public class LiteralStringPart {
-  public let uri: ExpressionUri<LiteralStringPart>
+public class HirLiteralStringPart {
+  public let uri: HirExpressionUri<HirLiteralStringPart>
```

Comment:
This won't work if `HirStringPart` doesn't implement `HirExpression`. Maybe we should inline string parts instead of using URIs for these?